src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -1226,19 +1005,73 @@ public int hashCode() {
         }
     }
 
-    private void logNodeMessage(FlowNode node) {
-        WorkflowConsoleLogger wfLogger = new WorkflowConsoleLogger(getListener());
-        String prefix = getBranchName(node);
-        if (prefix != null) {
-            wfLogger.log(String.format("[%s] %s", prefix, node.getDisplayFunctionName()));
-        } else {
-            wfLogger.log(node.getDisplayFunctionName());
+    /**
+     * Prints nodes as they appear (including block start and end nodes).
+     */
+    private final class NodePrintListener implements GraphListener.Synchronous {
+        @Override public void onNewHead(FlowNode node) {
+            NewNodeConsoleNote.print(node, getListener());
+        }
+    }
+
+    @SuppressWarnings("rawtypes")
+    @Override public AnnotatedLargeText getLogText() {
+        return LogStorage.of(asFlowExecutionOwner()).overallLog(this, !isLogUpdated());
+    }
+
+    // TODO log-related overrides pending JEP-207:
+
+    @Override public InputStream getLogInputStream() throws IOException {
+        // Inefficient but probably rarely used anyway.
```

Comment:
And comments like this make me realize that the log-handling rewrite that was supposed to deliver some great performance improvements may fall well short of that mark.

Reply:
My interest is in writes, not reads.

Reply:
Not much point of a log that nobody reads though, is there?  Anyway, we can see if it comes up as an issue.

Also, please do not mark this one "resolved" unless the code has been changed due to feedback -- it'll make it easier to address this if it come sup as a problem in the future.

Reply:
This particular method is rarely called AFAIK.